src/HID-APIs/MultiTouchAPI.h

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright (c) 2021 ilufang
+See the readme for credit to other people.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+// Include guard
+#pragma once
+
+/// Maximum amount of fingers supported
+#define HID_MULTITOUCH_MAXFINGERS 10
+/// Number of fingers in a single report
+#define HID_MULTITOUCH_REPORTFINGERS 2
+
+// A report will always be the same size, even if you report fewer fingers than
+// REPORTFINGERS. The unused finger entries will simply be zero. If more fingers
+// are present than REPORTFINGERS, multiple reports will be sent to report all
+// fingers. This is know as "Hybrid Mode" on MSDN. The number of supported
+// fingers identified by Windows will still be MAXFINGERS. More than MAXFINGERS
+// contacts may be ignored by Windows even with hybrid mode.
+
+#define _MT_STATE_INACTIVE 0
+#define _MT_STATE_CONTACT  1
+#define _MT_STATE_RELEASED 2
```

Comment:
What about those? What is the difference between those states and are they used by the windows client? If yes, why dont we expose these?

Nevertheless, please put them in an enum.

Reply:
No they are just for internal tracking. 0 and 1 just says if an entry is in use or not. 2 helps to identify a "release" action because windows expects that a released touch is reported once with tip-switch set off off after releasing, instead of just being silently excluded from the report.

Reply:
If my suggest API implementation will work, I think this tracking is no longer required.

Reply:
They are moved to hpp as an enum, but I think the current internal tracking cannot be further simplified. See below.